src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+                "Electricity : Electric Power Sector : Generation",
+                "Electricity : Electric Power Sector : Cumulative Retirements",
+                "Electricity : Electric Power Sector : Cumulative Planned Additions",
+                "Electricity : Electric Power Sector : Cumulative Unplanned Additions",
+                "Electricity : Electric Power Sector : Cumulative Total Additions",
+            ),
+        )
+        .pipe(_drop_aggregated_generation_data)
+        .pipe(_collect_totals)
+    )
+
+    assert set(sanitized.topic.unique()) == {"electricity"}
+    assert set(sanitized.subtopic.unique()) == {"electric_power_sector"}
+
+    trimmed = sanitized.drop(
+        columns=[
+            "topic",
+            "subtopic",
+            "units",
+        ]
+    )
+
+    unstacked = unstack(
+        df=trimmed,
+        eventual_pk=[
+            "report_year",
+            "model_case_eiaaeo",
+            "region",
+            "dimension",
+            "projection_year",
+        ],
+    )
+
+    # check that the totals all add up *before* decumulating, because
+    # decumulating introduces some small amount more floating point error.
+    ratio_close_additions_to_total, ratio_close_reported_calculated = (
+        check_totals_add_up(unstacked)
+    )
+    assert (
+        0.999 < ratio_close_additions_to_total <= 1.0
+    ), f"planned + unplanned vs. total: {ratio_close_additions_to_total}"
+    assert (
+        0.999 < ratio_close_reported_calculated <= 1.0
+    ), f"reported vs. calculated totals: {ratio_close_reported_calculated}"
+    # Now that the totals add up, we can drop the total_additions col
+    unstacked = unstacked.drop(columns="cumulative_total_additions")
+
+    decumulate_columns = [
+        "planned_additions",
+        "unplanned_additions",
+        "retirements",
+    ]
+    decumulated = _decumulate(unstacked, decumulate_columns=decumulate_columns)
+
+    # convert units: GW -> MW
+    for gw_col in [
+        "capacity",
+        "planned_additions",
+        "unplanned_additions",
+        "retirements",
+    ]:
+        decumulated[gw_col] *= 1000
+    # from billion KWh (1e12 Wh) to MWh (1e6 Wh)
+    decumulated.generation *= 1_000_000
+
+    renamed_for_pudl = (
+        decumulated.dropna(how="all")
+        .reset_index()
+        .pipe(lambda df: df.loc[(df.dimension != "total")])
+        .rename(
+            columns={
+                "capacity": "summer_capacity_mw",
+                "unplanned_additions": "summer_capacity_unplanned_additions_mw",
+                "planned_additions": "summer_capacity_planned_additions_mw",
+                "retirements": "summer_capacity_retirements_mw",
+                "generation": "gross_generation_mwh",
+                "region": "electricity_market_module_region_eiaaeo",
+                "dimension": "technology_description_eiaaeo",
+            }
+        )
+        .pipe(apply_pudl_dtypes, group="eiaaeo")
```

Comment:
i believe that the types of the tables will already be applied bc you are using `pudl_io_manager` (truly everyone does this bc its so helpful to make sure dtypes are type-able b4 trying to poop it out into the db but it is redundant bc types get applied during `enforce_schema`)

Reply:
I believe that too, I can take it out (but it was very helpful during development!)